### ISSUE TITLE
Bookmarks won't retain folder colour

### ISSUE THREAD

**Issue by airatin** [linuxmint/nemo#1071]
Hi,

Whenever I reboot Cinnamon or start Linux (Mint, 17.3), the folder colours won't appear in the bookmarks. However, they do appear once I click on that folder in Nemo. It would be nice if they appeared in the bookmarks right when I start the system, without having to click on them (or open a Nemo file directory to where these folders are located). Maybe I'm not explaining it well, so please see attached pictures (I added 2 custom colours of my own, in case you're wondering where those are from).

When I start up the system:
![before](https://cloud.githubusercontent.com/assets/16682521/12326031/4f539f52-ba9c-11e5-9dec-91b329d2d7b2.png)

After I open the file directory where the bookmarks (folders) are located, or click on them individually:
![after](https://cloud.githubusercontent.com/assets/16682521/12326043/5c50a47a-ba9c-11e5-8a97-9c8f16222628.png)

**Comment by mjenki03** [comment]
I think this is a dupe of https://github.com/linuxmint/nemo/issues/177

**Comment by airatin** [comment]
@mjenki03 I don't know if they're exactly duplicates since that one is about folder *icons* and not folder *colour*, but they're probably related. Interestingly enough, I just changed one of the folder icons to something else to test it out, and again it wouldn't show up until you went to the bookmark, just like in #177. Maybe a fix for folder colours can be easier than a fix for icons, but I doubt icons would be much harder.

It is unfortunate that it's been 3 years since #177 was filed, yet no progress has been made...

**Comment by airatin** [comment]
Hello, not sure how the GitHub system works but I still have this issue even after upgrading (my Cinnamon is nightly); sometimes when I boot the folder colours appear in the bookmarks, other times only a few colours appear, and other times no colours appear. I'm glad a response to this issue was made, but it didn't seem to solve the problem fully unfortunately. Should this issue be reopened?

**Comment by mtwebster** [comment]
Can you try to provide steps to reproduce what's still occurring? (I'm not doubting you, I just need a way to test any fix) - including, how the offending bookmarked locations are mounted/connected/etc..

I did do an extra update here: https://github.com/linuxmint/nemo/commit/d2f9c164a556bdc38561e8992df58cc0086e912a a few days after the initial fix - have you updated your nemo in the past 1-2 days?

**Comment by airatin** [comment]
Hi again, sorry for the late reply -- yes, I'm on the nightly release and I update my nemo/cinnamon whenever an update is available. I think I've narrowed down the source of the problem to my iPhone being mounted upon login; I did a series of tests and basically whenever my iPhone is mounted when I login, none of the folder colours appear in the bookmarks and I have to manually go to each folder to make the coloured folders appear again , but when it's not mounted (and the bookmarks do show the colours prior to shutdown, whether I make sure of this manually or whether they show up upon login by themselves since my iPhone wasn't mounted) and I login again, the bookmark colours appear fine. These are my more detailed results (Y = folder colours/icons appear in bookmarks, N = they don't appear in bookmarks):

- NO iPhone mounted upon login AND icons were restored before: Y
--- NO iPhone mounted upon login W/O restoring icons before: N (BUT Cinnamon launcher desklets and windows work as they should)
--- Restarting Cinnamon has no effect on these results
--- Ctrl+Alt+Backspace has no effect on these results

- YES iPhone mounted upon login: N (AND had to restart Cinnamon as launcher desklets (which have these coloured folders as their icons) don't show up, and window borders weren't right)
--- Unmount iPhone and then restart Cinnamon W/O restoring icons before: N
--- Unmount Phone and then restart Cinnamon AND icons were restored before: Y
--- NO unmount Phone and then restart Cinnamon W/O restoring icons before: N
--- NO unmount Phone and then restart Cinnamon AND icons were restored before: Y
--- Ctrl+Alt+Backspace and NO unmount W/O restoring icons before: N
--- Ctrl+Alt+Backspace and YES unmount W/O restoring icons before: N
--- Ctrl+Alt+Backspace and NO unmount AND icons were restored before: N
--- Ctrl+Alt+Backspace and YES unmount AND icons were restored before: N

**tl;dr:** mounted iPhones are somehow causing this to go wonky. What to do

**Comment by airatin** [comment]
Actually seems my hypothesis is wrong, it just comes up with no bookmark colour at all randomly. So, there is no fix to this issue at all and should be re-opened.

**Comment by xenopeek** [comment]
I've tried various ways to reproduce this but haven't managed to do so yet. If the issue happens again, please test against the latest Nemo version (3.6.5 currently), go through troubleshooting (see our [guide](http://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/)) and create a new issue if the responsible component proves to be Nemo.
